tests/thread_priority_inversion_mutex/main.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2017 Technische Universität Berlin
+ *               2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for testing priority inheritance for mutexes
+ *
+ * @author      Thomas Geithner <thomas.geithner@dai-labor.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+
+#include "thread.h"
+#include "mutex.h"
+#include "xtimer.h"
+
+#define TICK_LEN            (50UL * US_PER_MS)
+#define EXPECTED_RESULT     (4)
+
+static mutex_t _res_mtx = MUTEX_INIT;
+
+static char _stack_high[THREAD_STACKSIZE_MAIN];
+static char _stack_mid[THREAD_STACKSIZE_MAIN];
+static char _stack_low[THREAD_STACKSIZE_MAIN];
+
+static kernel_pid_t _pid_low;
+static kernel_pid_t _pid_mid;
+static kernel_pid_t _pid_high;
+
+static int _result = 0;
+static int _res_addsub = 1;
```

Comment:
could be done, be so far we don't really use local static variables in RIOT - though I can't recall the reasons. Poking the Internet I found quotes like

> Local statics may look useful, however they cause major problems when trying to port code to a multi-task/multi-threading environment, and should generally be avoided where possible.

Although this might no apply to this particular situation, IMHO it does not hurt either to have those variables in the compilation unit scope. On top I think declaring them like this makes it much easier to see the actual memory consumption of the module.

Reply:
Ok cool. Thanks for the info.